README.md

```diff
@@ -1 +1,412 @@
+5. Compose convenience initializers that call a designated initializer.
+
+## Initializer Methods
+
+We've discussed the use cases of the `alloc` and `init` method pairs in the past, but what are these methods actually doing? The `alloc` method is a class method that *allocates* a place in memory (RAM) to hold the instance variables that comprise an object of the given class. The `init` method is what actually populates that portion of memory with the relevant structure. Other initializers can be written which actually populate the structure with information (values) right from the start.
+
+We can compare these two roles to the phases of constructing a building: the `alloc` method is like selecting the ground that will be the building's lot, designating and preparing the place for it to go. The `init` method, however, is the phase of actually building the structure that's detailed in the architect's blueprint.
+
+The `init` method is actually just the basic initializer, called the "default initializer", but an initializer is any method that returns an `instancetype` and the compiler will require that the method name begins with `init...`. Methods from the `init` family can be set up to offer varying levels of pre-fabricated setup. 
+
+These "convenience initializers" and "designated initializers" can be seen like asking the construction firm to paint the house something other than white, or even to find an interior designer to provide furniture and to hang art on the walls. In such a case, the house is further along to becoming an actual home instead of just an empty free-standing structure that serves as the most basic form of a house.
+
+#### Initializing Read-Only Values
+
+Initializers are necessary for setting up read-only properties that cannot be written to from outside the class, such as in the case of creating an instance of `NSSortDescriptor`:
+
+```objc
+// incorrect: assigning to read-only properties
+
+NSSortDescriptor *sortByNameAsc = [[NSSortDescriptor alloc] init];
+sortByNameAsc.key = @"name";
+sortByNameAsc.ascending = YES;
+```
+
+![](https://curriculum-content.s3.amazonaws.com/ios-intro-to-objects-unit/readonly_error.png)
+
+The initializer allows a user of that class to provide an initial value for those read-only properties upon creating an instance:
+
+```objc
+// correct: using an initializer to give the properties initial values
+
+NSSortDescriptor *sortByNameAsc = [[NSSortDescriptor alloc] initWithKey:@"name"
+                                                              ascending:YES];
+```
+
+![](https://curriculum-content.s3.amazonaws.com/ios-intro-to-objects-unit/readonly_init.png)
+
+The usefulness of providing an initializer allows the user of the class to be assured that the instance will be set up holistically enough to perform its methods without unexpected behavior. It also allows us, the developer, to document the set up for the class.
```

Comment:
Hm... not quite sure what you mean here, or if this belongs here. Do you mean like... sort descriptors would be useless without a key & ascending stuff, so having an init method that takes those implicitly documents that those are required for it to even do anything useful? If so:

"Providing an initializer allows the class to ensure that instances have the information they need to behave appropriately from the get-go. It also allows us, the developer, to document requirements for setting up instances."